news/_posts/2020-12-09-pr-testing-on-prestashop.md

```diff
@@ -0,0 +1,179 @@
+
+### How do you choose which Pull Request should be tested first?
+
+This is a legitimate question that pops up regularly. It might be frustrating for a contributor to see their Pull Request blocked in “Waiting for QA” for weeks, while other PRs are validated and merged the same day they are created.
+
+The QA team uses the following criteria to sort Pull Requests (in order of importance):
+
+1. The branch on which Pull Request is based: the oldest [supported branch](https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/supported-branches/) has the highest priority. We will test `1.7.6.x` PRs before `1.7.7.x` PRs, and `1.7.7.x` PRs before `develop` PRs. PRs submitted to modules usually have the lowest priority.
+2. The priority of the issue the PR is linked to: A PR solving a “must-have” issue should be tested before a “nice-to-have”, and before an issue without any priority (remember that the priority is displayed with a label in Github).
+3. Pull Requests should be tested by submit date, from oldest to most recent.
+
+These are the theoretical rules to rank Pull Requests. However, sometimes the context forces the QA team to bypass it... For example, in addition to Pull Requests submitted to the Core, we also test Pull Requests submitted to PrestaShop's open source modules. If the number of Module Pull Requests waiting for QA is too high, we might test some of the oldest Modules Pull Requests in priority.
+
+And sometimes, the maintainers might ask us to test a PR that is required to unblock other tasks.
+
+### How long does it take to test a Pull Request?
+
+It highly depends on the complexity of the PR! A small fix, with a clear issue and a clear “How to Test” description can take as little as fifteen minutes to test, while a huge migration PR can require several days.
+
+If a PR is really complex, or if the feature being implemented is complex, there might be a second or even a third test by another QA specialist to validate it fully.
+
+### Why does my Pull Request have both the “Waiting for author” and “Waiting for QA” labels?
+
+If the QA analyst in charge asks a question on your PR (for example requiring more details on “How to test”), they will add the “Waiting for author” label, but without removing the “Waiting for QA” one. The latter is kept in order to be able to easily search for the PRs waiting for a response from its author.
+
+So if you see this label, it means more information is needed – check the last comments on the PR!
+
+# II. Test of a bug fix Pull Request
+
+PRs can be classified into 3 main categories (with a few special cases):
+
+- Bug fix
+- Improvement
+- Migration
+
+I will first focus on the bug fix Pull Request, since it’s the most common type there is, then detail the differences between a bug fix, an improvement and a migration PR.
```

Comment:
```suggestion
I will first focus on the bug fix Pull Request since it’s the most common type, then detail the differences between a bug fix, an improvement, and a migration PR.
```